### PR TITLE
Move to ristra inputs system.

### PR DESCRIPTION
**Pull request by timmah**
This moves the hydro and maire-hydro apps to the ristra input system. 

All the sample apps give the same answers as master (last week's master). 

One thing that is still sticky is the use of flecsale::math::array/vector. The templated ctor doesn't interact well with lua_result_t::as() method; compilation fails trying to convert lua_result_t to double. I'm assuming we'll decide on a single fixed-length array type from flecsi and flecsale to go into ristra, and use that going forward. For the moment, I explicitly convert flecsale::math::array to std::array.

### CODE REVIEW

File: apps/hydro/SimConfig.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// SimConfig.h
```
Comment (charest) [inline apps/hydro/SimConfig.h]: Please change this to match with the rest of the file names, i.e. no camel case.

File: apps/hydro/SimConfig.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+// Sep 01, 2017
```
Comment (charest) [inline apps/hydro/SimConfig.h]: Please add the standard header that all other files have

File: apps/hydro/base_problem.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                                    {"prefix", "shock_box_2d"},
+                                    {"suffix", "dat"},
+                                    {"mesh_type", "box"},
+                                    {"file", "zzxyzz"}};
+  hcs.set_registry<string_t>(string_reg);
+  // register array<size_t> defaults
+  reg<arrays_t> const arrays_reg = {{"dimensions", {10, 10}}};
+  hcs.set_registry<arrays_t>(arrays_reg);
+  // register array<real_t> defaults
+  reg<arrayr_t> const arrayr_reg = {{"xmin", {-0.5, -0.5}},
```
Comment (charest) [inline apps/hydro/base_problem.h]: Two dimensional-specific stuff should not appear here in the "hydro" folder.  It should be one level down.

File: apps/hydro/driver_new.h [hunk 4]
```diff
@@ -9,9 +9,13 @@
 #pragma once
```
Comment (charest) [inline apps/hydro/driver_new.h]: Missing header

File: apps/hydro/input_types.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
+// input_types.h
```
Comment (charest) [inline apps/hydro/input_types.h]: missing header

File: apps/hydro/tasks.h [hunk 6]
```diff
@@ -11,22 +11,40 @@
...
 #include "types.h"
-
+#include "input_types.h"
 // system includes
 #include <iomanip>
 
 namespace apps {
 namespace hydro {
 
+/**\brief Copy "vector" v to array a.
```
Comment (charest) [inline apps/hydro/tasks.h]: I don't think there is a strong enough case to provide our own array implementation here.  This is something that should be sorted out in the input class.   It was working fine with the previous lua_utils.

Comment (timmah) [reply apps/hydro/tasks.h]: Thanks, Marc, I'll get on these.

**Comment by timmah** [comment]
Changes addressed. CI will fail--no ristra.

**Comment by timmah** [comment]
Obsoleted by the closer integration with FleCSI--new PR soon.

**Pull request closed without merging**